sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/db/migration/V8520__Attachments.java

```diff
@@ -0,0 +1,177 @@
+package uk.ac.bbsrc.tgac.miso.db.migration;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
+
+public class V8520__Attachments implements JdbcMigration {
+
+  private static final Logger log = LoggerFactory.getLogger(V8520__Attachments.class);
+
+  private enum EntityType {
+    PROJECT("project", "Project", "projectId", "Project_Attachment"),
+    SERVICE_RECORD("servicerecord", "ServiceRecord", "recordId", "ServiceRecord_Attachment");
+    
+    private final String dirName;
+    private final String table;
+    private final String primaryKey;
+    private final String joinTable;
+    
+    private EntityType(String dirName, String table, String primaryKey, String joinTable) {
+      this.dirName = dirName;
+      this.table = table;
+      this.primaryKey = primaryKey;
+      this.joinTable = joinTable;
+    }
+
+    public String getDirName() {
+      return dirName;
+    }
+
+    public String getTable() {
+      return table;
+    }
+
+    public String getPrimaryKey() {
+      return primaryKey;
+    }
+
+    public String getJoinTable() {
+      return joinTable;
+    }
+  }
+
+  private Connection connection;
+  private String filesDir;
+  private Path basePath;
+
+  @Override
+  public void migrate(Connection connection) throws Exception {
+    log.info("Scanning existing attachments to save in database...");
+
+    this.connection = connection;
+    filesDir = getFilesDir();
+    if (LimsUtils.isStringEmptyOrNull(filesDir)) {
+      throw new IllegalStateException("MISO_FILES_DIR not set");
+    }
+    if (!filesDir.endsWith(File.separator)) {
+      filesDir += File.separator;
+    }
+    basePath = Paths.get(filesDir);
+
+    log.info("scanning directory {}", filesDir);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "scanning directory {}" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)